### PR TITLE
Update to Go 1.5

### PR DESCRIPTION
**Pull request by tamird**
The big day is here.

### CODE REVIEW

File: build/builder.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ image="cockroachdb/builder"
 
 function init() {
   docker build --tag="${image}" - <<EOF
-FROM golang:1.4.2
+FROM golang:1.5
```
Comment (petermattis) [inline build/builder.sh]: FYI, in order to get circleci to use this you'll have to push a new builder image and update the `tag` in `build/circle-deps.sh`.

**Comment by petermattis** [comment]
LGTM

File: build/builder.sh (hunk 1 above)
Comment (tamird) [inline build/builder.sh]: Thanks, doing this now.

**Comment by tschottdorf** [comment]
:+1: LGTM

File: build/builder.sh (hunk 1 above)
Comment (tamird) [inline build/builder.sh]: done

Comment (petermattis) [inline build/builder.sh]: Uh-oh, looks like the build-cache stuff isn't working properly. Could have sworn I put the go version number in there. You might need to rebuild with "clear-cache".

**Comment by tamird** [comment]
Almost there...

File: build/builder.sh (hunk 1 above)
Comment (petermattis) [inline build/builder.sh]: Heh: https://github.com/cockroachdb/build-cache/blob/master/pkg.go#L433

Comment (tamird) [inline build/builder.sh]: ha

Comment (tamird) [inline build/builder.sh]: shouldn't that still work correctly? we always `go install [...]/build-cache`, so the version it was compiled with should be incremented before the `build-cache restore` invocation. I'd expect that to correctly invalidate the cache

Comment (petermattis) [inline build/builder.sh]: I'm not sure. Something clearly wasn't working right. It's been a while since I added that TODO.

**Pull request merged**